akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -197,7 +195,7 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
           val eventSourcedSetup = new BehaviorSetup(
             ctx.asInstanceOf[ActorContext[InternalProtocol]],
             persistenceId,
-            emptyState,
+            Some(emptyState),
```

Comment:
isn't `EventSourcedBehaviorImpl` kept around? and it's a case class and therefore the constructor parameter `val emptyState` is kept around?

Reply:
Dang, indeed, missed that it's a case class.

Reply:
Actually, it's a case class but is there for an `apply(Context): Behavior` startup method, built using the state, I think there are no references kept, so maybe it can be garbage collected once started.

For the particular memory reachability with Java the class has no static method but create the Scala ESB on start for every instance. For Scala it is less likely to be a problem because of immutability and empty state being the same instance for every actor instance.

Reply:
Anyway maybe not a good solution for this because it comes at the cost of one extra object reference in Scala, per instance, where we'd currently expect a single immutable object being shared.